Orange/util.py

```diff
@@ -246,6 +248,13 @@ def __repr__(self):
                                  ', '.join('{}={!r}'.format(*pair)
                                            for pair in names_values))
 
+    @staticmethod
+    def __equal(obj1, obj2):
+        try:
+            return obj1.size == obj2.size and np.all(obj1 == obj2)
+        except AttributeError:
+            return np.all(obj1 == obj2)
```

Comment:
Because just one of the two objects might be an array_like. Comparing it to a non-array_like object, this still produces an array of bools. The above try case only handled both objects being array_like _and of same size_, hereby uncovering another flaw in my reasoning; already fixed up.